app/Models/LogDAO.php

```diff
@@ -19,12 +22,20 @@ public static function lines() {
 		return array_reverse($logs);
 	}
 
-	public static function truncate() {
-		file_put_contents(join_path(DATA_PATH, 'users', Minz_Session::param('currentUser', '_'), LOG_FILENAME), '');
+	public static function truncate(?string $logFileName = null) : void {
+		file_put_contents(self::currentUserLogPath($logFileName), '');
 		if (FreshRSS_Auth::hasAccess('admin')) {
 			file_put_contents(ADMIN_LOG, '');
 			file_put_contents(API_LOG, '');
 			file_put_contents(PSHB_LOG, '');
 		}
 	}
+
+	private static function currentUserLogPath(?string $logFileName = null): string {
+
+		if (is_null($logFileName)) {
```

Comment:
But as other falsy values such as `''` should also be rejected, I would better like a more general test:

```suggestion
		if ($logFileName != '') {
```

Reply:
Rememeber, behavior between PHP7 and PHP8 changes for non-strict comparisons.

Reply:
Testing for only null is exactly what is introducing a bug ;-)
Acceptations should be strict, rejections should be lenient.